File: experiment_jp_valid/j2pyraw/while__0__SPACE_OPTIMIZED_DP_SOLUTION_0_1_KNAPSACK_PROBLEM.java
class X { static int KnapSack ( int val [ ] , int wt [ ] , int n , int W ) { int mat [ ] [ ] = new int [ 2 ] [ W + 1 ] ; int i = 0 ; return ( n % 2 != 0 ) ? mat [ 0 ] [ W ] : mat [ 1 ] [ W ] ; }
 }

File: experiment/java-python/valid/j2pyrawpy/while__0__SPACE_OPTIMIZED_DP_SOLUTION_0_1_KNAPSACK_PROBLEM.py
#!/usr/bin/env python
""" generated source for module while__0__SPACE_OPTIMIZED_DP_SOLUTION_0_1_KNAPSACK_PROBLEM """
class X(object):
    """ generated source for class X """
    @classmethod
    def KnapSack(cls, val, wt, n, W):
        """ generated source for method KnapSack """
        mat = [None]*2
        i = 0
        return mat[0][W] if (n % 2 != 0) else mat[1][W]

